src/result_enumerator.rs

```diff
@@ -61,14 +61,13 @@ impl IWbemClassWrapper {
     }
 
     pub fn get_property(&self, property_name: &str) -> Result<Variant, WMIError> {
-        let name_prop =
-            WideCString::from_str(property_name)?;
+        let name_prop = BStr::from_str(property_name)?;
 
         let mut vt_prop: VARIANT = unsafe { mem::zeroed() };
 
         unsafe {
             (*self.inner.unwrap().as_ptr()).Get(
-                name_prop.as_ptr() as *mut _,
+                name_prop.as_bstr(),
```

Comment:
This change might technically be unnecessary.  This parameter is supposedly just a regular `LPCWSTR`, not a `BSTR`, but a `BSTR` can be passed to an `LPCWSTR` just fine (the prefixed length will simply be ignored).  However, between all the other APIs requiring BSTRs, and the example code of the method *also* using BSTRs, I erred on the side of catuion and paranoia (https://docs.microsoft.com/en-us/windows/win32/api/wbemcli/nf-wbemcli-iwbemclassobject-get#examples)

Reply:
I think it will be a bit nicer to have `as_lpcwstr()` for `BStr`, but to keep everything extra tidy. WDYT?

Reply:
Works for me!  https://github.com/ohadravid/wmi-rs/pull/28/commits/a8b8edb3b227a0053ba07625d41c757e31d4bbe6